### PR TITLE
ci: Refactor main pipeline

### PR DESCRIPTION
**Pull request by konrad-ohms**
The main pipeline still relied on AWS code build which we are going to decommission. This PR moves the build to our Concourse Cluster and also uses skopeo to more efficiently copy images.

Signed-off-by: Konrad Ohms <konrad.ohms@de.ibm.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -56,7 +55,8 @@ LABEL name="instana-agent-operator" \
       io.openshift.tags="" \
       io.k8s.description="" \
       com.redhat.build-host="" \
-      com.redhat.component=""
+      com.redhat.component="" \
+      org.opencontainers.image.authors="Instana, support@instana.com"
```
Comment (Milica-Cvrkota-IBM) [inline Dockerfile:59]: Would this be equivalent to the `MAINTAINER`?

File: ci/container-image-build-task.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  # docker load -i image/image.tar
+  # docker tag LOADED_IMAGE_ID instana-agent-docker-s390x:local
+  # docker run --rm -it --entrypoint sh instana-agent-docker-s390x:local # this won't run if you are not on an s390x machine.
+
+platform: linux
+image_resource:
+  type: registry-image
+  source:
+    repository: delivery.instana.io/int-docker-private-virtual/concourse/oci-build-task
+    tag: 0.11.0
```
Comment (Milica-Cvrkota-IBM) [inline ci/container-image-build-task.yml:13]: Should we use the latest here? Any reason for pinning the version?

File: ci/container-image-build-task.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# task that builds docker images with an option to specify a target platform.
```
Comment (Milica-Cvrkota-IBM) [inline ci/container-image-build-task.yml:1]: Should we add copy rights to this file too?

File: ci/create-tag-file.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+---
+#
+# (c) Copyright IBM Corp. 2021
```
Comment (Milica-Cvrkota-IBM) [inline ci/create-tag-file.yml:3]:
```suggestion
# (c) Copyright IBM Corp. 2024
```

File: ci/pipeline.yaml [hunk 5]
```diff
@@ -62,87 +56,43 @@ resources:
...
+  - name: agent-operator-image-amd64
     type: registry-image
     icon: docker
     source:
-      repository: gcr.io/instana-agent-qa/instana-agent-operator
-      tag: latest-amd64
-      username: _json_key
-      password: ((project-berlin-tests-gcp-instana-qa))
-  - name: operator-image-arm64
+      repository: delivery.instana.io/int-docker-agent-local/instana-agent-operator/dev-build
```
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:72]: I would vote to add a tag here as
```
tag: ((branch))
```
for all internal images. With this approach we would ensure that we can distinguish between different images in the artifactory UI. This is the approach we have in other PR pipelines

File: ci/pipeline.yaml [hunk 6]
```diff
@@ -151,6 +101,17 @@ resources:
       tag: latest
       username: iamapikey
       password: ((concourse-icr-containers-public.password))
+  - name: agent-operator-image-manifest-sha
+    type: key-value
+    icon: zip-disk
+  # - name: agent-operator-bundle-image
```
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:107]: should we remove this?

File: ci/pipeline.yaml [hunk 7]
```diff
@@ -354,52 +357,42 @@ jobs:
...
-            COMMIT_SHA: ((.:commit-sha))
+            GIT_COMMIT: ((.:git-commit))
             VERSION: ((.:operator-version))
-            JSON_KEY: ((project-berlin-tests-gcp-instana-qa))
             RED_HAT_REGISTRY_PASSWORD: ((redhat-container-registry-5e961c2c93604e02afa9ebdf.password))
             RED_HAT_REGISTRY_USERNAME: ((redhat-container-registry-5e961c2c93604e02afa9ebdf.user))
             ICR_PASSWORD: ((concourse-icr-containers-public.password))
             ICR_USERNAME: iamapikey
+            DELIVERY_INSTANA_IO_USER: ((delivery-instana-io-internal-project-artifact-read-writer-creds.username))
+            DELIVERY_INSTANA_IO_PW: ((delivery-instana-io-internal-project-artifact-read-writer-creds.password))
```
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:371]: This is just a suggestion to match the naming in the other pipelines
```suggestion
            ARTIFACTORY_USERNAME_INTERNAL: ((delivery-instana-io-internal-project-artifact-read-writer-creds.username))
            ARTIFACTORY_PASSWORD_INTERNAL: ((delivery-instana-io-internal-project-artifact-read-writer-creds.password))
```

File: ci/pipeline.yaml [hunk 8]
```diff
@@ -354,52 +357,42 @@ jobs:
...
             ARTIFACTORY_USERNAME: ((delivery-instana-io-release-project-artifact-read-writer-creds.username))
             ARTIFACTORY_PASSWORD: ((delivery-instana-io-release-project-artifact-read-writer-creds.password))
```
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:373]: This is just a suggestion to match the naming in the other pipelines
```suggestion
            ARTIFACTORY_USERNAME_RELEASE: ((delivery-instana-io-release-project-artifact-read-writer-creds.username))
            ARTIFACTORY_PASSWORD_RELEASE: ((delivery-instana-io-release-project-artifact-read-writer-creds.password))
```

File: ci/pipeline.yaml [hunk 9]
```diff
@@ -599,6 +520,7 @@ jobs:
             args:
               - -ceu
               - |
+                exit 1
```
Comment (Milica-Cvrkota-IBM) [inline ci/pipeline.yaml:523]: Let's remove this before merging

**Review by Milica-Cvrkota-IBM — CHANGES_REQUESTED** [review]
Thanks for the changes! Besides the comments, another note we discussed in slack is to put self-update before any other job

File: Dockerfile (hunk 1 above)
Comment (konrad-ohms) [reply Dockerfile:59]: yes, MAINTAINER is deprecated according to the Dockerfile reference: https://docs.docker.com/reference/dockerfile/#maintainer-deprecated

File: ci/container-image-build-task.yml (hunk 2 above)
Comment (konrad-ohms) [reply ci/container-image-build-task.yml:13]: Not sure, I tend to pin versions, but latest should work just fine as well. Probably latest would not hurt here. :+1:

File: ci/container-image-build-task.yml (hunk 3 above)
Comment (konrad-ohms) [reply ci/container-image-build-task.yml:1]: I'd say than we should them add to all pipeline files, but yes, I can add those.

File: ci/pipeline.yaml (hunk 6 above)
Comment (konrad-ohms) [reply ci/pipeline.yaml:107]: I was actually wondering that no olm bundle image was created, but apparently, we did not need it in that pipeline so far, just the bundle yaml files. So yeah, let me remove that.

**Review by Milica-Cvrkota-IBM — APPROVED** [review]
LGTM 👍

**Pull request merged**